src/cluster/autoscale.go

```diff
@@ -445,3 +452,21 @@ func getImageVersion(clusterID uint, cluster interface{}) map[string]string {
 
 	return selectedImageVersion
 }
+
+func setArchNodeSelector(clusterID uint, cluster interface{}) map[string]string {
```

Comment:
**Recommended**: IMO the function **SHOULD** take a strongly typed `cluster` and extract the cluster ID inside the body that way the input interface could be simplified to a cluster argument. Especially so because the cluster ID is merely a debug information in the function.